src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,20 @@
+package task;
+
+public class Deadline extends Task {
+    private String deadline;
+
+    public Deadline (String description, String deadline) {
+        super(description);
+        this.deadline = deadline;
+    }
+
+    @Override
+    public String getTaskDetails() {
+        return "[D][" + getStatusIcon() + "] " + description + " (by: " + deadline + ")";
+    }
```

Comment:
It is usually a practise to use `this.method()` or `this.variable` when that `method` or `variable` belongs to the Class.

This change can be made at other places as well :)

```suggestion
    public String getTaskDetails() {
        return "[D][" + this.getStatusIcon() + "] " + this.description + " (by: " + this.deadline + ")";
    }
```